### DISCUSSION TITLE
Where does Directus store thumbnails?

### DISCUSSION THREAD

**Discussion by powerseed** [directus/directus#13792] [Archive]
In https://docs.directus.io/reference/files/#requesting-a-thumbnail, the doc says that "If a requested thumbnail doesn't yet exist, it is dynamically generated and immediately returned", which sounds like Directus stores thumbnails that were previously requested. 

I am wondering where does Directus store them? I checked `directus_files` but they don't appear to exist there.

Thanks in advance!

**Comment by rijkvanzanten** [comment]
There's stored on the file system next to the original file 👍🏻 They're treated as variants of the original image, and won't show up as actual files in the file library

**Reply by powerseed** [reply]
Got it. Thanks so much!
